### DISCUSSION TITLE
Week 4 Status Updates

### DISCUSSION THREAD

**Discussion by roryeiffe** [roryeiffe/Fresh-Results#35] [General]
Respond to this thread with everything you've worked on from 4/8/21-4/15/21.

What did you work on this week?

What sort of technical blockers are you facing?

What are you going to work on next week?

What sort of new technologies did you learn, if any?

Put any links to commits/other contributions here.

Since we're getting close to the end of the semester, what are one or two more things that you want to get done before the semester is over?

Feel free to add more questions as you think of them!

**Comment by SerenaChen1** [comment]
# Serena Chen

## What did you work on this week?

I spent the beginning of this week working on a style guide, but we don't have much to document as of yet. Currently, I just have the colors we used in the guide. Then, I started working off of what Rory and Abdul did with message passing and using the effect hook. We have the spoiler censor colors changing now, but there's a bug with the way we are initializing it. It needs to get the color from local storage, but that's not working yet.

## What sort of technical blockers are you facing?

I was just having trouble understanding how message passing worked and what each script was used for. As mentioned above, we are blocked by a minor bug right now.

## What are you going to work on next week?

I'm going to continue working on changing the censor color. I'll also try to get the message passing to work for the spoiler threshold value.

## What sort of new technologies did you learn, if any?

I have a better understanding of message passing, local storage, and effect hooks now.

Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/commit/dcf570074d5f6448b6b6887286436c91f79c7e96
https://github.com/roryeiffe/Spoiler-Block/issues/28
https://github.com/roryeiffe/Spoiler-Block/commit/2fc99aa40a0ecabc56fbe7a6fed08984bc72db48

**Comment by roryeiffe** [comment]
## What did you work on this week?

This week, I met with the backend team to try to help import the data from a local JSON file into the extension. I also implemented placing colored blocks over pieces of text with spoilers. This was very exciting to work on since it is what we have been working towards this semester. It feels good to see all of our work pay off. Other than that, I fixed a few bugs, such as the extension itself not working as well as a bug where the user could still highlight and see the blocked text. I also added a state to the main App.js to make message passing to the background script easier. The state keeps track of the color and spoiler threshold and can keep track of any other state we need. Finally, I worked on getting message passing started. I got to an okay spot, but Abdul helped push it a little further and Serena is going to finish it up this week. 

## What sort of technical blockers are you facing?

None at the moment

## What are you going to work on next week?

Next week, Covenant and I are going to meet with the back-end team to try to get custom words imported into the extension. 

## What sort of new technologies did you learn, if any?

I learned how to pass messages from the extension to a background script, and then to the content script that runs on the web pages that the user visits. 

I used this [article](https://developer.chrome.com/docs/extensions/mv3/messaging/#port-lifetime) from chrome itself.

## Put any links to commits/other contributions here.

https://github.com/roryeiffe/Spoiler-Block/commit/6265fad47a4de6c0499aa81150aa40fe4bd5bebe
https://github.com/roryeiffe/Spoiler-Block/issues/16
https://github.com/roryeiffe/Spoiler-Block/commit/162d23d4a3ac2b9eef99e6fc81885823da3a01f9
https://github.com/roryeiffe/Spoiler-Block/commit/1d3ecf42140ccabcf77d06007e7236e270966ee6
https://github.com/roryeiffe/Spoiler-Block/issues/27
https://github.com/roryeiffe/Spoiler-Block/issues/28
https://github.com/roryeiffe/Spoiler-Block/pull/30
https://github.com/roryeiffe/Spoiler-Block/commit/86f835471422190efc7d28325ea82ee7b53a2281
https://github.com/roryeiffe/Spoiler-Block/commit/13f83faf0684f97ecdf847b2662e681e2ae0b37f
https://github.com/roryeiffe/Spoiler-Block/commit/a23cbd589466ac774eabce31034b43a63e8d9be8

## Since we're getting close to the end of the semester, what are one or two more things that you want to get done before the semester is over?

I think we've made a lot of good progress this semester. I would love to see Covenant's and Victoria's hard work integrated into the rest of the extension (custom words and pop-ups respectively).

**Comment by Sushiman33** [comment]
Respond to this thread with everything you've worked on from 4/8/21-4/15/21.

What did you work on this week?

- During this week I had gotten the app to block all the words from a JSON and made the fix to that code later on to load dictionary.json which was a file in the same directory as content.js

What sort of technical blockers are you facing?

- My inexperience in JavaScript is slowing me down in terms of progress as I do not know how things work off the top of my head and I have to do a bit more digging around to get things to work.

What are you going to work on next week?
- Going to fix the issue of the blocking words using a dictionary lookup

What sort of new technologies did you learn, if any?
-Learned how to use a fetch to grab a json from a local directory

Put any links to commits/other contributions here.
4eb90ca80523d8613e862b295705942252617edc
6669e6dc13f7c65c180fbec331dc9d2497635b89
60d2753c649d5a9e5d55f830165e207bf4eded2c

[Issue of loading from JSON](https://github.com/roryeiffe/Spoiler-Block/issues/23)
[Issue of replacing word from dictionary](https://github.com/roryeiffe/Spoiler-Block/issues/22)

Since we're getting close to the end of the semester, what are one or two more things that you want to get done before the semester is over?

- I want to get the loading from the dictionary working and fix any associated bugs with that.
- Add more words to the list of blocked words
- Utilize the weights of the word for some use

**Comment by TheCovenant** [comment]
### What did you work on this week:

This week I finished working on my 'keyword editor' component that enables custom 'potential spoiler' keywords to be added. I merged this with a PR. As of right now, all the functionality of the component is on the frontend but needs to be connected to the backend to be useful.

### What sort of technical blockers are you facing:

I'm not facing any significant technical blockers at this point. 

### What are you going to work on next week:

I will be meeting with the backend team to catch up on progress and find out the best way to integrate their work into my component in order to have the intended functionality it was created for.

### What sort of new technologies did you learn, if any:
None

## Since we're getting close to the end of the semester, what are one or two more things that you want to get done before the semester is over?

I think the 'keyword editing' feature is a really nice addition to the user experience of the extension. For that reason, I really hope to get it integrated with the backend before the semester ends.


### Links to Contributions:
https://github.com/roryeiffe/Spoiler-Block/pull/39/commits/9e9569d622cf1df2b86af6f4788e7ec3b139da2b
https://github.com/roryeiffe/Spoiler-Block/pull/39/commits/9d284d75286c2dcdda61814545b5e4a2e77af95e
